custom_metrics/pwa.js

```diff
@@ -69,17 +69,33 @@ const manifests = getEntriesForURLs(manifestURLs).map(([url, body]) => {
 const serviceWorkerInitiatedURLs = new Set(Array.from(serviceWorkerURLs).flatMap(getURLsInitiatedBy));
 const serviceWorkerInitiated = getEntriesForURLs(serviceWorkerInitiatedURLs);
 
-const workboxPattern = /(?:workbox:[a-z\-]+:[\d.]+|workbox\.[a-zA-Z]+\.?[a-zA-Z]*)/g;
 // We should use serviceWorkerInitiatedURLs here, but SW detection has some false negatives.
-const workboxInfo = response_bodies.filter(har => {
-  return workboxPattern.test(har.response_body);
-}).map(har => {
-  return [har.url, Array.from(har.response_body.matchAll(workboxPattern)).map(m => m[0])];
-});
+function getInfoForPattern(regexPattern) {
+  return response_bodies.filter(har => {
+    return regexPattern.test(har.response_body);
```

Comment:
Per the discussion, I'd suggest adding a condition here to filter out responses that contain non-SW code. For example, have a top-level const to define a pattern that disqualifies the response from being considered a SW and test for that here:

```suggestion
    return !disqualifyingSWPattern.test(har.response_body) && regexPattern.test(har.response_body);
```

Where `disqualifyingSWPattern` could be defined as something like:

```js
const disqualifyingSWPattern = /\bdocument\.\w+/g;
```

(not tested)

Reply:
This is great! I'm working on a new iteration to add more SW features to detect. I'll update the PR and then work on this condition to see if we can achieve the desired effect.

Reply:
Please, refer to the **Filtering "service worker only" activity** of [this comment](https://github.com/HTTPArchive/legacy.httparchive.org/pull/204#issuecomment-858998233), where I explained why this strategy didn't work.
Instead of that, I ended up adding an experimental field `serviceWorkerHeuristic`, which combines different conditions for features that should only appear in service workers. This is explained in [this comment](https://github.com/HTTPArchive/legacy.httparchive.org/pull/204#issuecomment-860066559).